torchquad/tests/helper_functions.py

```diff
@@ -44,124 +47,174 @@ def get_test_functions(dim, backend):
+                # over 3 integrand dims
+                Polynomial(np.array([[[0.0, 4.0], [8.0, 12.0]], [[16.0, 20.0], [24.0, 28.0]]]), [2.0], is_complex=False, backend=backend, integrand_dims=[2, 2, 2]),  # y = 2
+                Polynomial(np.array([[0.0, 0.0], [0.0, 0.0]]), [0, 1], is_complex=False, backend=backend, integrand_dims=[2, 2, 2]),   # y = x
+            ]
+        return res
+    elif integration_dim == 3:
+        res = [
             # Real numbers
             Polynomial(
-                48.0, [2.0], dim=3, is_complex=False, backend=backend
+                48.0, [2.0], integration_dim=3, is_complex=False, backend=backend, integrand_dims=1
             ),  # f(x,y,z) = 2
             Polynomial(
-                0, [0, 1], dim=3, is_complex=False, backend=backend
+                0, [0, 1], integration_dim=3, is_complex=False, backend=backend, integrand_dims=1
             ),  # f(x,y,z) = x + y + z
             # f(x,y,z) = x^2+y^2+z^2
-            Polynomial(8.0, coeffs=[0, 0, 1], dim=3, is_complex=False, backend=backend),
+            Polynomial(8.0, coeffs=[0, 0, 1], integration_dim=3, is_complex=False, backend=backend, integrand_dims=1),
             # e^x+e^y+e^z
             Exponential(
                 27 * (np.exp(3) - 1) / np.exp(2),
-                dim=3,
+                integration_dim=3,
                 domain=[[-2, 1], [-2, 1], [-2, 1]],
                 is_complex=False,
                 backend=backend,
+                integrand_dims=1
             ),
             Sinusoid(
                 24 * np.sin(1) ** 2,
-                dim=3,
+                integration_dim=3,
                 domain=[[0, 2], [0, 2], [0, 2]],
                 is_complex=False,
                 backend=backend,
+                integrand_dims=1
             ),
             # e^x+e^y+e^z
             Exponential(
                 1.756,
-                dim=3,
+                integration_dim=3,
                 domain=[[-0.05, 0.1], [-0.25, 0.2], [-np.exp(1), np.exp(1)]],
                 is_complex=False,
                 backend=backend,
+                integrand_dims=1
             ),
             #
             # Complex numbers
             Polynomial(
-                48.0j, [2.0j], dim=3, is_complex=True, backend=backend
+                48.0j, [2.0j], integration_dim=3, is_complex=True, backend=backend, integrand_dims=1
             ),  # f(x,y,z) = 2j
             Polynomial(
-                0, [0, 1.0j], dim=3, is_complex=True, backend=backend
+                0, [0, 1.0j], integration_dim=3, is_complex=True, backend=backend, integrand_dims=1
             ),  # f(x,y,z) = xj
             Polynomial(
-                8.0j, coeffs=[0, 0, 1.0j], dim=3, is_complex=True, backend=backend
+                8.0j, coeffs=[0, 0, 1.0j], integration_dim=3, is_complex=True, backend=backend, integrand_dims=1
             ),  # j*x^2+j*y^2+j*z^2
         ]
-    elif dim == 10:
+        if use_multi_dim_integrand:
```

Comment:
I only added tests that make sure the functionality works on a basic level since accuracy is handled by the other tests and this feature is basically just broadcasting.  I can definitely add some more rigorous accuracy checks though.

Reply:
do the tests pass on your machine though? They don't on my machine (with GPU) and also not on the CI. I'll attach the logs for more insight.

[6_Test with pytest.txt](https://github.com/esa/torchquad/files/10343094/6_Test.with.pytest.txt)

Reply:
I don't have a GPU so I was assuming numpy only - another reason this is a draft I had forgotten about.  I'll need to grab some time on a cluster.

Reply:
You can use the other frameworks on the CPU as well if that is easier. This conda environment sets up all of them: https://github.com/esa/torchquad/blob/main/environment_all_backends.yml

(the CI uses the same one)

Reply:
ah ok @gomezzz I just didn't install those because i was worried about my memory, but you're totally right. thanks.